### ISSUE TITLE
Signed releases

### ISSUE THREAD

**Issue by dillbyrne** [atom/atom#7635]
Could you please provide a signed checksums file or sign the files themselves for releases so we can cryptographically verify them.  Thanks

**Comment by matt40k** [comment]
Prob doesn't help, but the Windows build is code signed, so you can check the digital signature on File Properties. Not sure if other platforms have something similar.

**Comment by kevinsawicki** [comment]
The mac build is code signed as well

**Comment by dillbyrne** [comment]
Thanks for the information. I'm mainly concerned with the Linux releases. Since it could be a while before they make it into their respective repositories.

**Comment by matt40k** [comment]
Aren't the deb and rpm signed?

**Comment by dillbyrne** [comment]
If they are signed I could not find a public key anywhere in order to verify them.

**Comment by matt40k** [comment]
I think its built-in, it won't install if the signature if duff. Quick google turns up:
rpm -K 

It looks like it operates like the MS digital signature, the publisher apply to a certificate authority, gets verified, then they sign your certificate into a chain. The end-user trusts the certificate authority, who has in turn trusted the publisher (GitHub\Atom). So it looks like its all automatic - unless you compile from source.

I think what your thinking is the checksum for the release - which would be pretty pointless in terms of security, if someone hacked into GitHub and modified the release, to say, include spyware, they could also change the checksum. If you want to use the checksum for security, you'd want to have it listed on a totally separate site, kinda like they say with distro, download the image from which ever mirror, but get the checksum from the master site.

**Comment by dillbyrne** [comment]
Hello @matt40k  

RPM -K still requires a public key  http://www.rpm.org/max-rpm/s1-rpm-checksig-using-rpm-k.html

On Debian based systems  the verification methods available as far as I am aware are `debsig-verify` or `dpkg-sig` or you can do it manually with `gpg --verify` . 

All of the above options require a public key in addition to a signed file or signature file  

http://blog.packagecloud.io/eng/2014/10/28/howto-gpg-sign-verify-deb-packages-apt-repositories/

So if the binaries are signed then a public key still needs to be provided in order to verify this with the respective tools above.

So in short I'm looking for a GnuPG public key and a signature file (.asc/.pgp)  or signed binary with the signature built in.

**Comment by blaind** [comment]
+1

**Comment by siliconic** [comment]
bump. What is stalling a resolution to this issue?

**Comment by vancluever** [comment]
+1, and yeah, this is totally an issue. For the record, as it is not explicitly stated here, neither the .deb nor the .rpm are signed.

.deb:

```
$ debsigs --list atom-amd64.deb 
GPG signatures in atom-amd64.deb:
 *** NO ATTEMPT HAS BEEN MADE TO VERIFY THE LISTED SIGNATURES ***
```

.rpm:

```
$ rpm -qip atom.x86_64.rpm 
Name        : atom
Version     : 1.3.2
Release     : 0.1.fc21
Architecture: x86_64
Install Date: (not installed)
Group       : Unspecified
Size        : 212258338
License     : MIT
Signature   : (none)
Source RPM  : atom-1.3.2-0.1.fc21.src.rpm
Build Date  : Wed 16 Dec 2015 10:28:58 AM PST
Build Host  : 0e91de131a66
Relocations : /usr 
URL         : https://atom.io/
Summary     : A hackable text editor for the 21st Century.
Description :
A hackable text editor for the 21st Century.
```

I try and make it a rule to verify everything I download before installing these days, so it would be greatly appreciated if we could get signed packages.

**Comment by michaelgrahamevans** [comment]
How are the Linux packages STILL not being signed?

**Comment by juju4** [comment]
On macos, not all binaries are signed and that cause issue if using whitelisting like Google Santa (https://github.com/google/santa), especially during updates
mostly binaries of ShipIt, node and Electron crashpad_handler.

=> ideally can sign all with same certificate than Atom main binary

**Comment by natlibfi-arlehiko** [comment]
Any news on this?

**Comment by elsieholmes** [comment]
In fact, signing of .rpm packages is really simple and only takes --sign or --addsign options for rpmbuild and rpm respectively. All you need is a properly configured .rpmmacros file, GPG and a code signing key.

**Comment by kfiresmith** [comment]
This software really needs to live on a Github-hosted YUM repository that includes an RPM public key in ascii format like pretty much any other vendor-provided software in wide use.  At the very least sign the RPM before publishing it and dump an ASCII copy of the RPM signing public key somewhere with a hyperlink on the download page...
This is probably something that could be fixed in less than an hour of dev time by adding to your CICD pipeline that spits out new releases at https://atom.io/download...

**Comment by quantumpacket** [comment]
Been nearly 4yrs since issue was opened and still no signing is done for the Linux packages? What exactly is the roadblock on getting this closed? In this day and age something of this nature should have a much higher priority. :(

**Comment by wismill** [comment]
+1

**Comment by kfiresmith** [comment]
The maintainers appear to be either indifferent or lack the competence necessary to add a signing step to their RPM CICD.  I would have guessed it was mainly because RPMs are less popular than Debian/Ubuntu packages but from other linked issues it looks like they have had numerous other signing issues with other platforms.  I'm guessing the devs only run Ubuntu.

**Comment by quantumpacket** [comment]
I run Debian, but all they need to do is hash the files and sign the hash file and that's as good. AFAIK you can then force install the package if it complains about no signature, but who cares then because the signed hash will be verified.

They can at least sign the source so I can compile myself for whatever platform. I'm not going to randomly run unsigned code on my system. Especially when it's being hosted on a 3rd party site that is now run by a corporation that has documented partnering with numerous 3-letter agencies. [#1](https://www.theguardian.com/world/2013/jul/11/microsoft-nsa-collaboration-user-data), [#2](https://www.theverge.com/2013/6/6/4403868/nsa-fbi-mine-data-apple-google-facebook-microsoft-others-prism), [#3](https://www.independent.co.uk/news/edward-snowden-claims-microsoft-collaborated-with-nsa-and-fbi-to-allow-access-to-user-data-8705755.html), [#4](https://www.theregister.co.uk/2013/07/11/snowden_leak_shows_microsoft_added_outlookencryption_backdoor_for_feds/)

**Comment by kfiresmith** [comment]
The above proposal isn't a reasonable or useful solution.  Most people that care about code signing care because they have compliance controls at their operating environment that enforce RPM signatures.  Pretty much any corporate or government interpretation of various compliance frameworks enforce RPM signatures.  This causes what should be a hands off process (mirror an upstream repo) to become a manual process of pulling down RPMs, re-signing them, then re-hosting them in an internal mirror.  

The whole issue is just foolishness on the part of the Atom devs since they have already demonstrated that they can do this for Debian packages and the implementation for RPMs are every bit as trivial.

**Comment by snow8gaia** [comment]
I think my previous comment got deleted. 

Is this related to https://github.com/atom/atom/issues/18895? I was redirected here.

If it is can the title be updated? To make it clear for people looking for duplicates.

**Comment by lovermann** [comment]
Still now issues?

Cannot install it on openSUSE Leap 15.1.

> Linux linux-zsrx 4.12.14-lp151.28.44-default #1 SMP Fri Mar 20 18:20:20 UTC 2020 (dbf1aea) x86_64 x86_64 x86_64 GNU/Linux

> Error: INVALID:atom-1.46.0-0.1.x86_64 (file-eca764a8): Signature verification failed [6-File is unsigned]

**Comment by jlschuncke** [comment]
It's beyond mindboggling that a competent profession software organization seems incapable of the trivial task of signing their products.

I'm forced to disable the Atom repo on my various RPM-based linux machines until this is fixed.
